src/server/classroom/internal.controller.js

```diff
@@ -0,0 +1,56 @@
+const httpStatus = require('http-status');
+const Classroom = require('../models/classroom.model');
+const Messages = require('../models/message.model');
+const APIError = require('../helpers/APIError');
+
+/**
+ *
+ * @param {[*, *]} packet
+ * @param {import('socket.io').Socket} socket
+ * @param {import('socket.io').Server} io
+ */
+async function sendMessage(packet, socket, io) {
+  const [data, callback, meta] = packet;
+
+  if (!meta.invokerClassroom) return callback({ error: 'You are not in a classroom' });
+  const classroom = meta.invokerClassroom;
+
+  const message = await Messages.Message.create({
+    sender: meta.invoker.id,
+    type: 'text',
+    content: data.message,
+    classroom: classroom.id
+  });
+
+  classroom.messages.push(message);
+  await classroom.save();
```

Comment:
Whenever you modify the classroom database object, remember to invalidate cache
```suggestion
  await classroom.save();
  cachegoose.clearCache(`ClassroomById-${classroom.id}`);
```